src/idbstore.rs

```diff
@@ -16,9 +17,13 @@ impl IdbStore {
             None => return Ok(None),
         };
         let request = factory.open(name)?;
-
-        log!("request: {:?}", request);
-
+        let (sender, receiver) = oneshot::channel::<()>();
+        let success = Closure::wrap(Box::new(move || {
```

Comment:
This fails here, with:

```
repc!? cargo build
   Compiling replicache-client v0.1.0 (/Users/aa/work/roci/repc)
error[E0277]: expected a `std::ops::Fn<()>` closure, found `[closure@src/idbstore.rs:21:46: 24:10 sender:futures_channel::oneshot::Sender<()>]`
  --> src/idbstore.rs:21:37
   |
21 |           let success = Closure::wrap(Box::new(move || {
   |  _____________________________________^
22 | |             log!("onsuccess called!!1!");
23 | |             sender.send(());
24 | |         }) as Box<dyn Fn()>);
   | |__________^ expected an `Fn<()>` closure, found `[closure@src/idbstore.rs:21:46: 24:10 sender:futures_channel::oneshot::Sender<()>]`
   |
   = help: the trait `std::ops::Fn<()>` is not implemented for `[closure@src/idbstore.rs:21:46: 24:10 sender:futures_channel::oneshot::Sender<()>]`
   = note: wrap the `[closure@src/idbstore.rs:21:46: 24:10 sender:futures_channel::oneshot::Sender<()>]` in a closure with no arguments: `|| { /* code */ }
   = note: required for the cast to the object type `dyn std::ops::Fn()`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `replicache-client`.

To learn more, run the command again with --verbose.
```

From some research it appears that the error message is bad, and actual problem is that my closure is `FnOnce`: https://github.com/rustwasm/wasm-bindgen/issues/1269.

However, attempting to wrap `sender` in an option and taking it inside the closure doesn't help.